### ISSUE TITLE
Two users are unable to upload files

### ISSUE THREAD

**Issue by amyehodge** [sul-dlss/happy-heron#1934]
Cheryl Butner reports on https://sdr.stanford.edu/works/3033. She is attempting to replace a file on an existing item.
File is 2.26GB. Error message: Unable to upload file due to: Error storing "Intertidal.zip". Status: 502. Delete the file and try again. 

Katherine Poggensee reports on https://sdr.stanford.edu/works/3900/edit. She is attempting to upload 700-800MB. Error message: Unable to upload file due to: Error storing [name of file]. Status: 0. Delete the file and try again." This is the same error message that Cheryl got the first time. Katie has been trying this from off campus with the VPN on.

I'm concerned about these errors and would like someone to look into them. We haven't had upload errors on hydrus in probably years for files of these sizes.

**Comment by amyehodge** [comment]
@hannahfrost

**Comment by jcoyne** [comment]
I suspect that `Status: 502` was when we were deploying updates to prod, they occurred at about the same time.  

Is it possible that some of these errors are now just more visible? Now you get a message, but in hydrus it would just not load the next page and the user wouldn't see any error.

**Comment by amyehodge** [comment]
People who were unable to get files uploaded -- error message or not -- did contact us. I don't think the problem is that they are reporting this more frequently now. Regardless, it should be able to handle file uploading as well as Hydrus or we will have a lot of manual work on our hands with Globus. These reports are for content well within the capacity of Hydrus.

**Comment by jcoyne** [comment]
Are the upload problems consistent for these users?  Are they unable to upload any files of any size?

**Comment by jcoyne** [comment]
We could look at `chunking: true, retryChunks: true` (in Dropzone) https://codecalamity.com/upload-large-files-fast-with-dropzone-js/  But I don't think that activestorage DiskService supports this, so we'd have to write our own storage service.  This is trivial to support for s3...but we aren't using cloud services. Maybe we could get ops to install MinIO?

**Comment by amyehodge** [comment]
Notes from testing with Cheryl Butner. 
I just got on zoom with Cheryl and watched her upload files. She is on a Mac using Chrome from her home (not the marine station).
She was able to upload a 98kb Word document using both the drag and drop and the "choose files" options. She then immediately tried to drag and drop the same intertidal.zip file as above and when the green bar was all the way to the right and it looked finished, she got the error message with "Status:0". 

I'll be doing testing with Janet Sakai tomorrow morning -- third person to report upload problems.

**Comment by amyehodge** [comment]
I received the file from Cheryl and was able to upload it to her object this morning. It took about 5-6 minutes on our fiber connection here. For the record, I'm on a Mac using Chrome and I had the VPN connected. I used the drag and drop option.

**Comment by amyehodge** [comment]
Details from a third user, Janet Saiki, who reported issues with uploading a file of about 250MB to https://sdr.stanford.edu/works/4638. She had trouble yesterday doing this from home but was successful this morning from Lathrop.

Machine: PC (Stanford machine, so the same one both times)
Browser: Chrome
VPN: connected
Option: Tried the drag and drop (twice) + choose files button (once)
Connection to internet: she's at her brother's house using some wifi connector provided by SUL IT to his internet service. I was unable to obtain info on what type of service that is (doesn't seem like she knows). She did say that it seemed slower than usual yesterday.

**Comment by justinlittman** [comment]
I wonder if it is worth testing the uploads that are problematic with H2 with Hydrus?

**Comment by amyehodge** [comment]
We could give Cheryl temporary access to Hydrus again to see if she can upload her file there from her house. That's possibly useful? She was able to upload it to Google Drive to send it to me. How different is that?

**Comment by jcoyne** [comment]
@amyehodge are you able to download the file from Google and upload it to h2?

**Comment by amyehodge** [comment]
@jcoyne Yes. I was able to download it from Google and upload it to H2 without any trouble. Upload took 5-6 minutes on my fiber connection. Hers errored out after about 20 minutes with the progress bar fully green.

**Comment by amyehodge** [comment]
Details from testing with Katherine Poggensee.

Machine: PC
Browser: Chrome
VPN: yes
Option: drag & drop OR choose files? Tried both
Connection to internet: wifi on probably cable, but her upload speed to Google Drive was quite slow
Item: https://sdr.stanford.edu/works/3900
File size: 772MB, able to upload a 333MB

I watched her upload the 772MB file again and file. She then put it in a Drive folder and I downloaded it and uploaded it tot he same item without any trouble. File: c3d_Day2.zip

**Comment by amyehodge** [comment]
Cheryl Butner's speed test results:
![CherylSpeedTest.png](https://images.zenhubusercontent.com/5e9e06e81c23039800e23efc/87ebf5a9-fb25-4085-8f92-57f28acd664d)

**Comment by jcoyne** [comment]
<img width="730" alt="Screen Shot 2021-08-17 at 11 18 01 AM" src="https://user-images.githubusercontent.com/92044/129763050-c0adc188-6c48-4a94-84e0-8bcfcfa25c8b.png">

My upload speed is 30x faster than that, which might be why this isn't a problem for me.

**Comment by amyehodge** [comment]
Ditto. Here's mine:
![Screen Shot 2021-08-17 at 12.20.13 PM.png](https://images.zenhubusercontent.com/5e9e06e81c23039800e23efc/4adc2829-8c7a-4f1d-ac7a-733e43a91df7)

**Comment by amyehodge** [comment]
From Katherine: 
"Right now it's 23.8/14.5 download/upload. My internet seems to be especially slow this morning, so I can check again this afternoon too."

**Comment by amyehodge** [comment]
Speed test notes from Janet:
"Download results: 17.1 Mpbs
Upload results: 0.8 Mpbs"

That upload speed is crazy slow.

**Comment by amyehodge** [comment]
I just used Microsoft Remote Desktop to connect to one of the machines in the GIS lab on campus. I downloaded two files of 3.5 and 4.4GB sent to me by Cheryl Butner and attempted to upload them to her items via that machine. The green bar filled in pretty quickly (I didn't time it), but at the end I got an error:

![Screen Shot 2021-08-18 at 4.43.25 PM.png](https://images.zenhubusercontent.com/5e9e06e81c23039800e23efc/22d6554a-1e60-43c5-b829-3c4d7e43f997)

That remote machine is directly wired into the campus network, not on wifi.

**Comment by jcoyne** [comment]
@amyehodge it's important that we distinguish the two separate problems here.  One is "Status 502" which means the application isn't responding, and "Status 0" which means a network error.

**Comment by jcoyne** [comment]
I saw these lines in the Apache web server error log that may correspond:
```
[Wed Aug 18 13:42:42.306666 2021] [proxy_http:error] [pid 32655] (70007)The timeout specified has expired: [client 171.67.128.59:57465] AH01102: error reading status line from remote server localhost:3000, referer: https://sdr.stanford.edu/works/3034/edit
[Wed Aug 18 13:42:42.331013 2021] [proxy:error] [pid 32655] [client 171.67.128.59:57465] AH00898: Error reading from remote server returned by /rails/active_storage/disk/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaDdDam9JYTJWNVNTSWhaRzloTVRWc2EzVnhhR2hqY0hCMGJtRTNZelpxZURVd2RYUjRaZ1k2QmtWVU9oRmpiMjUwWlc1MFgzUjVjR1ZKSWlGaGNIQnNhV05oZEdsdmJpOTRMWHBwY0MxamIyMXdjbVZ6YzJWa0Jqc0dWRG9UWTI5dWRHVnVkRjlzWlc1bmRHaHNLd2dCSEw0R0FRQTZEV05vWldOcmMzVnRTU0lkYjI4d1MxYzNVVkJpZWtkWWNXSnpielJCYzNCdlFUMDlCanNHVkRvUmMyVnlkbWxqWlY5dVlXMWxPZ3BzYjJOaGJBPT0iLCJleHAiOiIyMDIxLTA4LTE4VDIwOjQ1OjAzLjA5MFoiLCJwdXIiOiJibG9iX3Rva2VuIn19--02e26bccf5a57b83fc61a854e8df1bc2d0176f3b, referer: https://sdr.stanford.edu/works/3034/edit
```

**Comment by jcoyne** [comment]
I believe that ops can fix the "502" problem by setting a larger value for `ProxyTimeout`. I believe it is currently set to 2 minutes.  The reason we can't just turn it up to infinity is that we want connections to timeout if there's really no hope of them concluding and then the resource can be reused.  If all the resources are used, then nobody can use the server.  Would 5 minutes be a reasonable amount of time to wait for a file upload?

**Comment by amyehodge** [comment]
Ah. I haven't been distinguishing between the two issues all the time, you're right.

**Comment by amyehodge** [comment]
@jcoyne 10 minutes would be better. If people can't get their files up we'll have to do it for them on the back end, and we're trying to get out of that business, not deeper in.

**Comment by jcoyne** [comment]
Can you see if they are able to upload the one that gave the 502 error today?

**Comment by jcoyne** [comment]
@amyehodge Let's try 5 first and see if we run into more errors. I'm concerned that turning it to 10 means we could prevent the webserver from handling requests from other users.

**Comment by amyehodge** [comment]
@jcoyne Can it only handle one request at a time?

I'm testing this now.

**Comment by amyehodge** [comment]
I was able to connect remotely to that desktop on campus and get both of the large files uploaded to their respective items. I then attempted this twice from my fiber connection here in VT. It failed the first time and I neglected to note which error message I got. When I tried it again (after shutting off the webinar I was watching), the upload succeeded. Thanks upping the time. It definitely helped.
